From foo@baz Tue Sep  1 20:12:58 PDT 2009
Date: Tue, 01 Sep 2009 20:12:58 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: hv: warn the world of a bug in the release function

All device release functions need to do something, if not, it's a bug.
By merely providing an "empty" release function, it gets the kernel to
shut up, but that's not solving the problem at all.  Stick a big fat
WARN_ON(1); in there to get people's attention.


Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/hv/vmbus_drv.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/staging/hv/vmbus_drv.c
+++ b/drivers/staging/hv/vmbus_drv.c
@@ -965,6 +965,11 @@ Desc:	Final callback release of the vmbu
 static void vmbus_bus_release(struct device *device)
 {
 	DPRINT_ENTER(VMBUS_DRV);
+	/* FIXME */
+	/* Empty release functions are a bug, or a major sign
+	 * of a problem design, this MUST BE FIXED! */
+	dev_err(device, "%s needs to be fixed!\n", __func__);
+	WARN_ON(1);
 	DPRINT_EXIT(VMBUS_DRV);
 }
 
